app/post.go

```diff
@@ -1174,3 +1174,131 @@ func (a *App) MaxPostSize() int {
 
 	return maxPostSize
 }
+
+func (a *App) countMentionsFromPost(user *model.User, post *model.Post) (int, *model.AppError) {
+	channel, err := a.GetChannel(post.ChannelId)
+	if err != nil {
+		return 0, err
+	}
+
+	if channel.Type == model.CHANNEL_DIRECT {
+		// Return -1 to indicate that the mention count should be equal to the post count
+		return -1, nil
```

Comment:
since this is a sentinel value, shouldn't we name it for clarification?

Reply:
As mentioned [here](https://github.com/mattermost/mattermost-server/pull/11966#discussion_r322968740), I couldn't find a good place to put a named value for this because of how the `app` and `sqlstore` packages are isolated from one another. I don't think an extra parameter helps this since it would just be another nameless argument that might not be understood by whoever is calling this.

I've made sure there's a comment explaining the purpose of this value wherever it's used for now, but I'm open to any further suggestions on how we might improve it.

Reply:
I added a `MentionAllPosts` constant for this to the `store` package. I know we've added things like this to the `model` package before, I feel like the `store` is the right place for it. If we ever need to export that value to a user of the plugin API or the web API, we can move the constant out of the `store` package.